### PR TITLE
feat: Sapt stage 1

### PR DESCRIPTION
**Pull request by outsidethecode**

### CODE REVIEW

**Comment by outsidethecode** [comment]
I've added an initial skeleton code to implement the SATP. This includes creating the relevant proto files, the requires client and services provided by the gateway. The focus was on the interactions 1.3, 1.4, 2.2 and 2.4. This code review is to ensure the draft code is readable and maintainable so that similar approach will be taken to implement the remaining functions.

File: .github/workflows/test_weaver-fabric-fabric-data-sharing.yaml [hunk 1]
```diff
@@ -0,0 +1,1044 @@
+# Copyright IBM Corp. All Rights Reserved.
```
Comment (sandeepnRES) [inline .github/workflows/test_weaver-fabric-fabric-data-sharing.yaml:1]: This file is for what purpose?

File: .github/workflows/test_weaver-fabric-fabric-asset-transfer.yaml [hunk 2]
```diff
@@ -0,0 +1,367 @@
+# Copyright IBM Corp. All Rights Reserved.
```
Comment (sandeepnRES) [inline .github/workflows/test_weaver-fabric-fabric-asset-transfer.yaml]: If you want to use this file for `satp_test`, may be call it `test_weaver-satp.yaml`

File: weaver/core/relay/Cargo.toml [hunk 3]
```diff
@@ -11,16 +11,23 @@ path = "src/main.rs"
...
+
+[[bin]]
+name = "satp_client"
+path = "src/satp_client.rs"
+
 [[bin]]
 name = "client-tls"
 path = "src/client_tls.rs"
 
+[[bin]]
```
Comment (sandeepnRES) [inline weaver/core/relay/Cargo.toml]: this is duplicate entry as line 15-17 also points to same file `src/satp_client.rs`

File: weaver/core/relay/src/services/network_service.rs [hunk 4]
```diff
@@ -373,42 +449,177 @@ impl Network for NetworkService {
...
+            request
+        );
+        let conf = self.config_lock.read().await.clone();
+        let network_asset_transfer = request.into_inner().clone();
+        let transfer_commence_request: TransferCommenceRequest =
+            derive_transfer_commence_request(network_asset_transfer.clone());
+        let request_id = transfer_commence_request.session_id.to_string();
+        // TODO refactor
+        let request_logged: Result<Option<sled::IVec>, crate::error::Error> =
+            log_request_in_local_sapt_db(&request_id, &network_asset_transfer, conf.clone());
```
Comment (sandeepnRES) [inline weaver/core/relay/src/services/network_service.rs]: `sapt` typo, also check other places too.

Comment (outsidethecode) [reply weaver/core/relay/src/services/network_service.rs]: fixed.

File: weaver/core/relay/Cargo.toml (hunk 3 above)
Comment (outsidethecode) [reply weaver/core/relay/Cargo.toml]: Good spot. Fixed.

File: .github/workflows/test_weaver-fabric-fabric-asset-transfer.yaml (hunk 2 above)
Comment (outsidethecode) [reply .github/workflows/test_weaver-fabric-fabric-asset-transfer.yaml]: So maybe test_weaver-fabric-fabric-satp.yaml

File: .github/workflows/test_weaver-fabric-fabric-data-sharing.yaml (hunk 1 above)
Comment (outsidethecode) [reply .github/workflows/test_weaver-fabric-fabric-data-sharing.yaml:1]: In this file I thought it is better to have the test done between two fabric networks only as a starting point.

**Pull request merged**